sqlmesh/core/engine_adapter/databricks.py

```diff
@@ -1,13 +1,112 @@
 from __future__ import annotations
 
+import logging
+import typing as t
+
+import pandas as pd
+from sqlglot import Dialect, exp
+
 from sqlmesh.core.engine_adapter.spark import SparkEngineAdapter
 from sqlmesh.core.schema_diff import SchemaDiffer
+from sqlmesh.utils import classproperty
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core.engine_adapter._typing import DF, PySparkSession
 
+logger = logging.getLogger(__name__)
 
-class DatabricksSparkSessionEngineAdapter(SparkEngineAdapter):
+
+class DatabricksEngineAdapter(SparkEngineAdapter):
     DIALECT = "databricks"
     SCHEMA_DIFFER = SchemaDiffer(
         support_positional_add=True,
         support_nested_operations=True,
         array_element_selector="element",
     )
+
+    def __init__(
+        self,
+        connection_factory: t.Callable[[], t.Any],
+        dialect: str = "",
+        sql_gen_kwargs: t.Optional[t.Dict[str, Dialect | bool | str]] = None,
+        multithreaded: bool = False,
+        **kwargs: t.Any,
+    ):
+        super().__init__(
+            connection_factory,
+            dialect,
+            sql_gen_kwargs,
+            multithreaded,
+            **kwargs,
+        )
+        self._spark: t.Optional[PySparkSession] = None
+
+    @classproperty
+    def can_access_spark_session(cls) -> bool:
+        from sqlmesh import runtime_env
+
+        if runtime_env.is_databricks:
+            return True
+        try:
+            from databricks.connect import DatabricksSession  # type: ignore
+
+            return True
+        except ImportError:
+            return False
+
+    @property
+    def _use_spark_session(self) -> bool:
+        from sqlmesh import runtime_env
+
+        if runtime_env.is_databricks:
+            return True
+        return self.can_access_spark_session and {
+            "databricks_connect_server_hostname",
+            "databricks_connect_access_token",
+            "databricks_connect_cluster_id",
+        }.issubset(self._extra_config)
+
+    @property
+    def is_spark_session_cursor(self) -> bool:
+        from sqlmesh.engines.spark.db_api.spark_session import SparkSessionCursor
+
+        return isinstance(self.cursor, SparkSessionCursor)
+
+    @property
+    def spark(self) -> PySparkSession:
```

Comment:
Can we raise here if `_use_spark_session` is False? I think that would be the best place to raise a clear error message if our expectations about using the spark session hasn't been met. Additionally I think this method is something user can access in a python model.